crates/egui/src/widgets/image.rs

```diff
@@ -769,3 +785,28 @@ pub fn paint_texture_at(
         }
     }
 }
+
+fn get_index(ctx: &Context, uri: &str) -> usize {
+    let now = ctx.input(|i| Duration::from_secs_f64(i.time));
+    let durations: Option<Arc<Vec<Duration>>> =
+        ctx.data(|data| data.get_temp(Id::new(format!("{uri}-index"))));
```

Comment:
Maybe you could use `Id::new((uri, "index"))` to avoid allocating a `String` each time? Or maybe even an enum type to generate image data Ids.

```rs
#[derive(Clone, Copy, PartialEq, Eq, Hash)]
pub enum ImageDataId {
    Index,
}

impl ImageDataId {
    #[inline]
    pub fn id(self, url: &str) -> Id {
        Id::new((self.type_id(), self, url))
    }
}
```

And then

```rs
ImageDataId::Index.id(url)
```

So there is less risk of bugs due to using different Ids in different parts of the code. Maybe that's overkill though.

Reply:
@YgorSouza I didn't see the point of using a enum. Is there a reason why enum & not a struct?